### PR TITLE
Add BigQuery sink example

### PR DESCRIPTION
**Pull request by RVRhub**
Add BigQuery Sink based on JSON format.
To generate data and send it to BigQuery you need to have a service account key file.

Then you can run the following command:
```bash
bin/datafaker_gen -f json -n 2 -sink bigquery
```

### CODE REVIEW

File: datafaker-gen-examples/datafaker-gen-bigquery/src/test/java/net/datafaker/datafaker_gen/sink/SchemaGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        assertEquals("id", fields.get(2).getName());
+        assertEquals("lastname", fields.get(3).getName());
+        assertEquals("address", fields.get(1).getName());
+
+        FieldList subFields = fields.get(1).getSubFields();
+        assertEquals(3, subFields.size());
+        assertEquals("country", subFields.get(0).getName());
+        assertEquals("city", subFields.get(1).getName());
+        assertEquals("street address", subFields.get(2).getName());
+    }
```
Comment (snuyanzin) [inline datafaker-gen-examples/datafaker-gen-bigquery/src/test/java/net/datafaker/datafaker_gen/sink/SchemaGeneratorTest.java]: lt's use assertj instead

File: datafaker-gen-examples/datafaker-gen-bigquery/src/test/java/net/datafaker/datafaker_gen/sink/BigQuerySinkTest.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class BigQuerySinkTest {
+
+    @Test
+    void testRunWhenWrongConfiguration() {
+        String serviceAccount = getClass().getClassLoader().getResource("test_service_account.json").getPath();
+
+        BigQuerySink bigQuerySink = new BigQuerySink();
+        assertThrows(RuntimeException.class, () -> {
```
Comment (snuyanzin) [inline datafaker-gen-examples/datafaker-gen-bigquery/src/test/java/net/datafaker/datafaker_gen/sink/BigQuerySinkTest.java]: assertj

File: datafaker-gen-examples/pom.xml [hunk 3]
```diff
@@ -17,5 +17,6 @@
 
     <modules>
         <module>datafaker-gen-rabbitmq</module>
+        <module>datafaker-gen-bigquery</module>
```
Comment (snuyanzin) [inline datafaker-gen-examples/pom.xml:19]: let's keep abc order

File: datafaker-gen-examples/datafaker-gen-bigquery/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  bigquery:
+    batchsize: 1
+    project_id: _project_id_
+    dataset: datafaker
+    table: users
+    service_account: path_to_service_account.json
+    create_table_if_not_exists: true
+    max_outstanding_elements_count: 100
+    max_outstanding_request_bytes: 10000
+````
```
Comment (snuyanzin) [inline datafaker-gen-examples/datafaker-gen-bigquery/README.md]: it seems one tick should be removed

File: datafaker-gen-examples/datafaker-gen-bigquery/src/main/java/net/datafaker/datafaker_gen/sink/BigQueryConnectionConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        this.maxOutstandingElementsCount = maxOutstandingElementsCount;
+        this.maxOutstandingRequestBytes = maxOutstandingRequestBytes;
+    }
+
+    public static BigQueryConnectionConfiguration fromConfig(Map<String, ?> config) {
+        final Credentials credentials;
+
+        String serviceAccount = (String) config.get("service_account");
+        try (FileInputStream fis = new FileInputStream(serviceAccount)) {
+            if (config.get("test") != null) {
```
Comment (snuyanzin) [inline datafaker-gen-examples/datafaker-gen-bigquery/src/main/java/net/datafaker/datafaker_gen/sink/BigQueryConnectionConfiguration.java]: `test` why so strange name?

File: datafaker-gen-examples/datafaker-gen-bigquery/src/main/java/net/datafaker/datafaker_gen/sink/BigQueryStreamWriter.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+                                FlowControlSettings.newBuilder()
+                                        .setMaxOutstandingElementCount(configuration.getMaxOutstandingElementsCount())
+                                        .setMaxOutstandingRequestBytes(configuration.getMaxOutstandingRequestBytes())
+                                        .build());
+        return builder
+                .setExecutorProvider(FixedExecutorProvider.create(Executors.newScheduledThreadPool(100)))
+                .setChannelProvider(
+                        BigQueryWriteSettings.defaultGrpcTransportProviderBuilder()
+                                .setKeepAliveTime(org.threeten.bp.Duration.ofMinutes(1))
+                                .setKeepAliveTimeout(org.threeten.bp.Duration.ofMinutes(1))
```
Comment (snuyanzin) [inline datafaker-gen-examples/datafaker-gen-bigquery/src/main/java/net/datafaker/datafaker_gen/sink/BigQueryStreamWriter.java]: ideally these parameters should be moved to config

File: datafaker-gen-examples/datafaker-gen-bigquery/src/main/java/net/datafaker/datafaker_gen/sink/BigQueryConnectionConfiguration.java (hunk 5 above)
Comment (RVRhub) [reply datafaker-gen-examples/datafaker-gen-bigquery/src/main/java/net/datafaker/datafaker_gen/sink/BigQueryConnectionConfiguration.java]: I've added an additional property responsible for the type of credential loader. Now, we can support the following approaches for loading credentials:
  - **service_account**: This starts with ServiceAccountCredential. - This is Default loader.
  - **application_default**: This is the default approach from Google. This means that credential loader following chain: ComputeCredential (if not available, then) -> UserCredential (if not available, then) -> ServiceAccountCredential.
  - **Test**: This simply creates a dummy OAuth2Credentials.

**Pull request merged**